### PR TITLE
fix(test-utils): improve test setup

### PR DESCRIPTION
**Pull request by emmenko**
While I was implementing tests for an application, I encountered some problems and missing things that can be provided by the `test-utils` out of the box.

I recommend a review-per-commit.

### CODE REVIEW

File: packages/application-shell/src/from-core/modal-container/modal-container.js [hunk 1]
```diff
@@ -3,12 +3,26 @@ import React from 'react';
...
+// element, used to render the modal content.
+// Apparently though, when the <Modal> component unmounts between tests, it
+// tries to remove the child elements within the parent selector
+// ("portals-container") which may have been cleaned up by another test,
+// resulting in the <Modal> to throw an NotFoundError.
+// To solve this, we can simply use "document.body" as the parent selector
+// instead of a specific element that will be cleaned up, resulting in
+// console errors (even though the test passes). We only need to to this in
+// test environment.
+const getParentSelector = () =>
```
Comment (dferber90) [inline packages/application-shell/src/from-core/modal-container/modal-container.js]: I would call this function `parentSelector` as `parentSelector={parentSelector}` makes more sense to me than `parentSelector={getParentSelector}`.

File: packages/application-shell/src/test-utils/test-utils.js [hunk 2]
```diff
@@ -160,29 +160,33 @@ const render = (
...
   {
-    // Consumers of renderWithRedux can use
-    //   { store: createReduxStore({ requestsInFlight: null, .. }) }
-    // to pass an initial state to Redux.
-    store = createReduxStore(),
+    // Pass an initial state that will be used to rehydrate the redux store.
+    // Possible state values are:
+    // - requestsInFlight: []
+    // - notifications: []
+    preloadedState,
```
Comment (dferber90) [inline packages/application-shell/src/test-utils/test-utils.js]: With this change, we gain a simpler API to write a test, but we lose the ability to pass in an arbitrary store. Which means we can't replace the middleware for our tests anymore. I'm not sure whether we should already limit the API that much. Did we learn that nobody ever needs to pass in a configured store?

Comment (emmenko) [reply packages/application-shell/src/test-utils/test-utils.js]: I thought about it but I don't know when we would be needing to replace the `store`. But I can revert this change if you prefer so.

Comment (dferber90) [reply packages/application-shell/src/test-utils/test-utils.js]: I generally like to keep the API open until the use-cases are clear. Then we can restrict the API to force people in the right direction. I can't make a good call here at the moment.

Are all stores configured the same for all applications? If yes, then it makes sense to configure the store in a similar way for our tests and we might not need to enable injecting middleware. It's still an assumption though.

I'd keep it open for now until we've seen more cases using it. If they all end up being the same (using preloaded state only), we can limit the API.

Comment (emmenko) [reply packages/application-shell/src/test-utils/test-utils.js]: I reverted the change

**Pull request merged**